admin/services/school-impersonation.service.js

```diff
@@ -0,0 +1,44 @@
+'use strict'
+const R = require('ramda')
+
+const schoolDataService = require('../services/data-access/school.data.service')
+const schoolImpersonationDfeNumberValidator = require('../lib/validator/school-impersonation-dfe-number-validator')
+
+const schoolImpersonationService = {}
+
+/**
+ * Validate and assign school to helpdesk session
+ * @param {object} user
+ * @param {string} dfeNumber
+ * @returns {Object}
+ */
+schoolImpersonationService.validateImpersonationForm = async (user, dfeNumber) => {
+  let validationError
+  validationError = schoolImpersonationDfeNumberValidator.isDfeNumberEmpty(dfeNumber)
+  if (validationError.hasError()) {
+    return validationError
+  }
+  let school
+  try {
+    school = await schoolDataService.sqlFindOneByDfeNumber(dfeNumber)
+  } catch (error) {}
+  validationError = schoolImpersonationDfeNumberValidator.isDfeNumberValid(school)
+  if (validationError.hasError()) {
+    return validationError
+  }
+  user.School = school.dfeNumber
+  user.schoolId = school.id
+  user.timezone = school.timezone
```

Comment:
It seems to me that we lose the knowledge that the user is impersonating.  We should be able to know:
* that the user is impersonating
* the true role for the user

Can you confirm?

Reply:
@jon-shipley knowledge so far about if the user is already impersonating is not needed and the role itself is exposed already in the authentication part.